src/lib/log/log_sys.c

```diff
@@ -0,0 +1,35 @@
+/* Copyright (c) 2018, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * \file log_sys.c
+ * \brief Setup and tear down the logging module.
+ **/
+
+#include "orconfig.h"
+#include "lib/subsys/subsys.h"
+#include "lib/log/escape.h"
+#include "lib/log/log.h"
+#include "lib/log/log_sys.h"
+
+static int
+init_logging_subsys(void)
+{
+  init_logging(0);
+  return 0;
+}
+
+static void
+shutdown_logging_subsys(void)
+{
+  logs_free_all();
+  escaped(NULL);
```

Comment:
Originally, the `escaped(NULL)` is before the free all. Will it impact anything?

Reply:
It shouldn't; the escaped(NULL) is just meant to free the buffer held by escaped().

Reply:
Ack.